src/amo/pages/AddonInfo/index.js

```diff
@@ -144,6 +146,24 @@ export class AddonInfoBase extends React.Component<InternalProps> {
       infoType,
     } = this.props;
 
+    if (errorHandler.hasError()) {
+      log.warn(`Captured API Error: ${errorHandler.capturedError.messages}`);
+
+      // 401 and 403 for an add-on lookup is made to look like a 404 on purpose.
+      // See https://github.com/mozilla/addons-frontend/issues/3061
+      if (
+        errorHandler.capturedError.responseStatusCode === 401 ||
+        errorHandler.capturedError.responseStatusCode === 403 ||
+        errorHandler.capturedError.responseStatusCode === 404
+      ) {
```

Comment:
This seems new?

Reply:
It is. When I was testing locally I discovered that this page looks ugly if the API returns a 404. We should treat it the same as the other pages that render a `NotFoundPage` in these cases.